### PR TITLE
Arc - attempt to fix bean types for beans with raw type superclasses

### LINKED ISSUE #30571: fix the set of bean types when one of supertypes is a raw type
### Describe the bug

If a bean class has a _raw type_ as one of its supertypes, ArC does not construct the set of bean types correctly. The JLS says:

> The superclass types (respectively, superinterface types) of a raw type are the erasures of the superclass types (superinterface types) of the named class or interface.

But ArC does not reflect this.

### Expected behavior

Erasures of supertypes of a raw type should be present in the set of bean types.

### Actual behavior

Not erased supertypes of a raw type are present in the set of bean types.

### How to Reproduce?

CDI TCK, test `org.jboss.cdi.tck.tests.definition.bean.BeanDefinitionTest#testRawBeanTypes`.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by manovotn**
Fixes #30571

This fixes the TCK scenario (added as a new test) and should comply with JLS expectations as per linked issue.

### CODE REVIEW

**Review by Ladicek — APPROVED** [review]

**Comment by manovotn** [comment]
CI run looks fine except Win run which seems to have encountered an unrelated timeout/cancellation.
@mkouba please add a review and if you're fine with it, let's merge.

File: independent-projects/arc/processor/src/test/java/io/quarkus/arc/processor/TypesTest.java [hunk 1]
```diff
@@ -93,6 +94,17 @@ public void testGetTypeClosure() throws IOException {
         assertDoesNotThrow(
                 () -> verifyEagleTypes(Types.getClassBeanTypeClosure(index.getClassByName(DotName.createSimple(Eagle.class)),
                         dummyDeployment)));
+
+        // raw type bean
+        Set<Type> rawBeanTypes = Types.getClassBeanTypeClosure(index.getClassByName(DotName.createSimple(MyRawBean.class)),
+                dummyDeployment);
+        assertEquals(rawBeanTypes.size(), 5);
+        assertTrue(rawBeanTypes.contains(Type.create(DotName.createSimple(MyRawBean.class.getName()), Kind.CLASS)));
```
Comment (mkouba) [inline independent-projects/arc/processor/src/test/java/io/quarkus/arc/processor/TypesTest.java:102]: Just nitpicking but this looks like a good candidate for a util method...?
```java
private void assertContainsType(Class<?> clazz, Set<Type> rawBeanTypes) {
   assertTrue(rawBeanTypes.contains(Type.create(DotName.createSimple(clazz.getName()), Kind.CLASS)));
}
```

**Review by mkouba — APPROVED** [review]

File: independent-projects/arc/processor/src/test/java/io/quarkus/arc/processor/TypesTest.java (hunk 1 above)
Comment (manovotn) [reply independent-projects/arc/processor/src/test/java/io/quarkus/arc/processor/TypesTest.java]: Fair enough, introduced util method :)

**Comment by manovotn** [comment]
Addressed the comment and rebased onto `main`.
Once the CI runs, this can be merged.

**Pull request merged**